game-engine/src/main/scala/za/co/entelect/challenge/game/contracts/map/CarGameMap.scala

```diff
@@ -189,4 +191,103 @@ class CarGameMap(players: util.List[Player], mapGenerationSeed: Int, lanes: Int,
   def placeObjectAt(lane: Int, blockNumber: Int, mapObject: Int) = {
     blocks.find(x => x.getPosition().getLane() == lane && x.getPosition().getBlockNumber() == blockNumber).get.mapObject = mapObject;
   }
+
+  def stageFuturePosition(stagedPosition: StagedPosition) ={
+    stagedFuturePositions = stagedFuturePositions.appended(stagedPosition);
+  }
+
+  def resolvePlayerCollisions(): Boolean = {
+    val player1StagedPosition = stagedFuturePositions.find(x => x.getPlayer().getGamePlayerId() == 1).get
+    val player1FuturePosition = player1StagedPosition.getNewPosition()
+
+    val player2StagedPosition = stagedFuturePositions.find(x => x.getPlayer().getGamePlayerId() == 2).get
+    val player2FuturePosition = player2StagedPosition.getNewPosition()
+
+    val playersFuturePositionsAreSame = (player1FuturePosition.getLane() == player2FuturePosition.getLane()) && (player1FuturePosition.getBlockNumber() == player2FuturePosition.getBlockNumber())
+
+    val player1DroveIntoPlayer2 = (player1StagedPosition.getOldPosition().getLane() == player2StagedPosition.getOldPosition().getLane()) && //started off in the same lane
+      (player1StagedPosition.getOldPosition().getBlockNumber() < player2StagedPosition.getOldPosition().getBlockNumber()) && //player 1 was behind
+      (player1FuturePosition.getBlockNumber() >= player2FuturePosition.getBlockNumber()) && //player 1 ended up in front
+      (player1FuturePosition.getLane() == player2FuturePosition.getLane()) //in the same lane => player 2 got ridden over
+
+    val player2DroveIntoPlayer1 = (player2StagedPosition.getOldPosition().getLane() == player1StagedPosition.getOldPosition().getLane()) && //started off in the same lane
+      (player2StagedPosition.getOldPosition().getBlockNumber() < player1StagedPosition.getOldPosition().getBlockNumber()) && //player 2 was behind
+      (player2FuturePosition.getBlockNumber() >= player1FuturePosition.getBlockNumber()) && //player 2 ended up in front
+      (player2FuturePosition.getLane() == player1FuturePosition.getLane()) //in the same lane => player 1 got ridden over
+
+    val isCollisionFromBehind = player1DroveIntoPlayer2 || player2DroveIntoPlayer1
+
+    val isCollision = playersFuturePositionsAreSame || isCollisionFromBehind
+
+    if(!isCollision)
+    {
```

Comment:
I think this brace belongs in c# code 😉

Reply:
😝 lol, my bad. Habits die hard

Reply:
Will boy-scout as I go along 👍 and fix this one